client/packages/system/src/Patient/VaccinationCards/DetailView/DetailView.tsx

```diff
@@ -12,30 +11,63 @@ import {
   NothingHere,
   LocaleKey,
   useFormatDateTime,
+  useRowStyle,
+  useTheme,
+  StatusCell,
 } from '@openmsupply-client/common';
 import {
   usePatientVaccineCard,
   VaxCardData,
 } from '../../api/hooks/usePatientVaccineCard';
 
+const useStyleRowsByStatus = (rows: DoseRowData[]) => {
+  const { setRowStyles } = useRowStyle();
+  const theme = useTheme();
+
+  useEffect(() => {
+    if (!rows) return;
+
+    setRowStyles(
+      rows.filter(row => !!row.dateGiven).map(row => row.id),
+      {
+        backgroundColor: `${theme.palette.background.success} !important`,
+      }
+    );
+  }, [rows]);
```

Comment:
I guess I could set Row Styles for the remaining rows to explicitly set their background to white, which would solve the alternating colour problem described above? (Unless there's a better way?)

Reply:
Done, but I had to add an additional flag to the `setRowStyles` method to not reset the row styles, because when you do two sequential `setRowStyles()`, the styles get reset in between and only the last one sticks.

7ed8975781